source/common/upstream/outlier_detection_impl.h

```diff
@@ -219,13 +286,34 @@ class DetectorImpl : public Detector, public std::enable_shared_from_this<Detect
 
   void onConsecutive5xx(HostSharedPtr host);
   void onConsecutiveGatewayFailure(HostSharedPtr host);
+  void onConsecutiveLocalOriginFailure(HostSharedPtr host);
   Runtime::Loader& runtime() { return runtime_; }
   DetectorConfig& config() { return config_; }
 
   // Upstream::Outlier::Detector
   void addChangedStateCb(ChangeStateCb cb) override { callbacks_.push_back(cb); }
-  double successRateAverage() const override { return success_rate_average_; }
-  double successRateEjectionThreshold() const override { return success_rate_ejection_threshold_; }
+  double
+  successRateAverage(DetectorHostMonitor::SuccessRateMonitorType monitor_type) const override {
+    return std::get<0>(success_rate_nums_.at(monitor_type));
+  }
+  double successRateEjectionThreshold(
+      DetectorHostMonitor::SuccessRateMonitorType monitor_type) const override {
+    return std::get<1>(success_rate_nums_.at(monitor_type));
+  }
+
+  /**
+   * This function returns tuple of double values for success rate outlier detection. The pair
+   * contains the average success rate of all valid hosts in the cluster and the ejection threshold.
+   * If a host's success rate is under this threshold, the host is an outlier.
+   * @param success_rate_sum is the sum of the data in the success_rate_data vector.
+   * @param valid_success_rate_hosts is the vector containing the individual success rate data
+   *        points.
+   * @return std::tuple<double, double>.
```

Comment:
Unless you switch to using a struct with named fields, please make sure we're clearly documenting what the two doubles in the tuple actually are.

Reply:
Using simple struct now with meaningful data members' names and comments.